### PR TITLE
Add support for absolute opeartion

### PR DESCRIPTION
**Pull request by PariksheetPinjari909**
This PR adds support for absolute operation.

### CODE REVIEW

File: python/tvm/intrin.py [hunk 1]
```diff
@@ -285,6 +285,22 @@ def trunc(x):
...
+    ----------
+    x : Expr
+        Input argument.
+
+    Returns
+    -------
+    y : Expr
+        The result.
+    """
+    return call_pure_intrin(x.dtype, "fabs", x)
```
Comment (tqchen) [inline python/tvm/intrin.py]: Please also add support dispatching integer values, implement this function in the c++ side.

```c++
Expr tvm::abs(Expr x) {
   if (x.type().is_int()) {
      return select(x >= make_zero(x.type()), x, -x);
   } else if (x.type().is_float()) {
     return call_pure_intrin("fabs")...
   } else if (x.type().is_uint()) {
      return x;
   }
}
```

File: topi/include/topi/elemwise.h [hunk 2]
```diff
@@ -35,6 +35,7 @@ TOPI_DECLARE_UNARY_OP(floor);
 TOPI_DECLARE_UNARY_OP(ceil);
 TOPI_DECLARE_UNARY_OP(round);
 TOPI_DECLARE_UNARY_OP(trunc);
+TOPI_DECLARE_UNARY_OP(fabs);
```
Comment (tqchen) [inline topi/include/topi/elemwise.h]: topi function name needs to be consistent with numpy, absolute.

File: python/tvm/intrin.py (hunk 1 above)
Comment (tqchen) [reply python/tvm/intrin.py]: implementation should go to here https://github.com/dmlc/tvm/blob/master/include/tvm/ir_operator.h#L61

**Comment by PariksheetPinjari909** [comment]
@masahi  @zhreshold Can you please review.

**Comment by PariksheetPinjari909** [comment]
@tqchen thanks for review comments, i will work on it and update.

**Comment by zhreshold** [comment]
Is there an `abs` alias for `absolute`? Or did you mean only floating point is supported since you are using `fabs`

**Comment by PariksheetPinjari909** [comment]
@zhreshold numpy has aliased 'np.abs' and 'np.absolute'. Now I have kept same name 'abs' to be consistent with numpy, TF, mxnet, onnx.

**Comment by PariksheetPinjari909** [comment]
@tqchen I have addressed review comments, pls review again.

File: include/tvm/ir_operator.h [hunk 3]
```diff
@@ -71,6 +70,24 @@ inline Expr pow(Expr x, Expr y) {
   return ir::Call::make(x.type(), "pow", { x, y }, ir::Call::PureIntrinsic);
 }
 
+/*!
+ * \brief Calculate absolute value of x, elementwise
+ * \param x The input data
```
Comment (tqchen) [inline include/tvm/ir_operator.h]: add \return

File: python/tvm/intrin.py [hunk 4]
```diff
@@ -285,6 +285,22 @@ def trunc(x):
...
```
Comment (tqchen) [inline python/tvm/intrin.py]: need to call abs c++ function here, instead of calling intrinsic

Comment (PariksheetPinjari909) [reply python/tvm/intrin.py]: Can you guide me, how can i make a call from tvm python to tvm cpp function? Can you give me any example?

Comment (srkreddy1238) [reply python/tvm/intrin.py]: tvm.abs !!

Comment (tqchen) [reply python/tvm/intrin.py]: you will need to register functions, and they will appear on the registered namespace. Checkout e.g. https://github.com/dmlc/tvm/blob/master/src/api/api_ir.cc#L76 try register abs under make.abs, and call it from there

**Comment by PariksheetPinjari909** [comment]
@tqchen I have registered absolute op as make.abs in api_ir.cc. Pls check.

**Comment by tqchen** [comment]
Thanks @srkreddy1238 @zhreshold for code reviews and @PariksheetPinjari909 for contributing the code, this is now merged!

**Pull request merged**